Sources/JSON.swift

```diff
@@ -48,9 +48,7 @@ public protocol JSONCollectionType {
 
 extension JSONCollectionType {
     public func jsonData() throws -> Data {
-        guard let jsonCollection = self as? AnyObject else {
-            throw MarshalError.typeMismatchWithKey(key:"", expected: AnyObject.self, actual: self.dynamicType) // shouldn't happen
-        }
+        let jsonCollection = self
```

Comment:
Xcode was complaining about using self when it should be optional so this is what is left. I'm not 100% sure this is correct so if I'm missing something please let me know and I'll update it.